### ISSUE TITLE
daemon sending wrong/old IP-address on update

### ISSUE THREAD

**Issue by RedCookieLP** [troglobit/inadyn#436]
After getting the client configured and leaving it running for a while, i noticed that my DDNS-service didn't update to the new address.

I was already running the daemon with debug-output (since I had issues while setting it up), so I checked the logs and saw that it correctly got the new IPv6-address, but then proceeded to send the old IPv6-address to the server.

This is the log of the daemon (private stuff censored):
![grafik](https://github.com/troglobit/inadyn/assets/45202005/eb637fa5-839f-40b5-bc91-31c2c3fd87fe)

The yellow highlighted part is part of the new IPv6, the magenta one is part of the old IPv6.
So it's clear that it's sending the old IP, for some weird reason.
That's also why the server responds with `nochg`, because it already knows the old IP.

If it helps, here's my config's `ddns-path` and `checkip-command`:
```
ddns-path       = "/nic/update?myipv6=%i&hostname="
checkip-command = "ip -6 addr list scope global $iface | grep -v \"deprecated\" | grep -v \" fd\" | sed -n 's/.*inet6 \\([0-9a-f:]\\+\\).*/\\1/p'"
```
The `&hostname=` at the end is intentional, even if nothing is there. The DDNS-service-support told me, that their API ignores the `hostname`-parameter anyways, so they don't care if it has a proper value or not. But it's still there because the hostname is automagically appended... sometimes.

**Comment by troglobit** [comment]
What is the version of your inadyn and how did you install it? Sometimes bugs are fixed in newer versions and sometimes distributions add their own patches. Impossible for us here upstream to know.

**Comment by RedCookieLP** [comment]
I am using inadyn 2.8.1 on Ubuntu 22.10, installed using apt

**Comment by troglobit** [comment]
I was afraid of that, you should report issues with packaged software with your distribution.  They often carry very old releases, v2.8.1 was released 2021-01-31, the latest release is v2.11.0 and a v2.11.1 is coming out very soon. 

In this case I believe your problem is a duplicate of issue #367, fixed in 1f6ae75 for v2.9.1.

**Comment by RedCookieLP** [comment]
Ok, i manually upgraded to 2.11.0 and using `custom` it *seems* to be working fine so far.
But i noticed that `ipv6@inwx.com` is now a supported provider.
(I am trying to use INWX's DDNS with IPv6... don't know why i censored it before)

I configured it for that using this config... (the `checkip-command` is the same as before)
```
provider ipv6@inwx.com {
        ssl             = true
        username        = ...
        password        = ...
        hostname        = ...
        checkip-command = "ip -6 addr list scope global $iface | grep -v \"deprecated\" | grep -v \" fd\" | sed -n 's/.*inet6 \\([0-9a-f:]\\+\\).*/\\1/p'"
}
```
...but it sent the IPv6 in the wrong parameter (IPv6-address in `myip`-parameter, not `myipv6`), so INWX just discarded it and used my NAT-IPv4 instead. (<-- this was confirmed by INWX's support)
Is this a bug or did i misconfigure `inadyn` here?

**Comment by troglobit** [comment]
Great to hear custom works better in v2.11.0!

In v2.11.0 a LOT of new providers were added, courtesy of the DD-WRT project, but I have not had the ability to test and verify any of the changes.  

It would be greatly helpful if you could debug this issue.  It sounds like a bug and simply adjusting the URL the plugin uses and recompile should be enough.

**Comment by troglobit** [comment]
Any progress?

**Comment by troglobit** [comment]
I made a speculative fix which will be part of v2.12.0, which pending testing will be out any day now.
